.circleci/config.yml

```diff
@@ -134,35 +134,64 @@ jobs:
-              aws configure set aws_secret_access_key $AWS_SECRET_ACCESS_KEY
-              aws configure set default.region $AWS_REGION
+                  if [ -n "<< pipeline.parameters.SECRETS >>" ]; then
+                    echo "Downloading and decoding secrets..."
 
-              # Download secrets file from S3
-              aws s3 cp << pipeline.parameters.SECRETS >> ~/project/encrypted_secrets.json
+                    # Configure AWS CLI
+                    aws configure set aws_access_key_id $AWS_ACCESS_KEY_ID
+                    aws configure set aws_secret_access_key $AWS_SECRET_ACCESS_KEY
+                    aws configure set default.region $AWS_REGION
 
-              # Decrypt the secrets
-              DECRYPTED_JSON=$(node ~/project/decrypt.js "$(cat ~/project/encrypted_secrets.json)")
+                    # Download secrets file from S3
+                    aws s3 cp << pipeline.parameters.SECRETS >> ~/project/encrypted_secrets.json
 
-              echo "$DECRYPTED_JSON" | jq -r 'to_entries | map("export \(.key)=\(.value|@sh)") | .[]' > ~/project/decrypted_secrets.txt
-              source ~/project/decrypted_secrets.txt
+                    # Decrypt the secrets
+                    DECRYPTED_JSON=$(node ~/project/decrypt.js "$(cat ~/project/encrypted_secrets.json)")
 
-              # Clean up
-              rm ~/project/encrypted_secrets.json
-            fi
+                    echo "$DECRYPTED_JSON" | jq -r 'to_entries | map("export \(.key)=\(.value|@sh)") | .[]' > ~/project/decrypted_secrets.txt
+                    source ~/project/decrypted_secrets.txt
 
-            if [ -n "<< pipeline.parameters.ENVIRONMENT_VARIABLES >>" ]; then
-              echo "Processing environment variables..."
-              echo "<< pipeline.parameters.ENVIRONMENT_VARIABLES >>" | sed 's/^/export /' > ~/project/environment_variables.txt
-              source ~/project/environment_variables.txt
-            fi
+                    # Clean up
+                    rm ~/project/encrypted_secrets.json
+                  fi
+
+                  # Fetch environment variables from the One Click Server
+                  ENV_VARS_URL="<< pipeline.parameters.ONE_CLICK_SERVER_URL >>/projects/<< pipeline.parameters.PROJECT_ID >>/env-vars"
+                  ENV_VARS=$(curl -s -H "x-admin-key: $ADMIN_KEY" $ENV_VARS_URL)
+                  if [ $? -ne 0 ] || [ -z "$ENV_VARS" ]; then
+                    echo "Failed to fetch environment variables"
+                    exit 1
+                  fi
+
+                  echo "Fetched Environment Variables: $ENV_VARS"
+
+                  # Extract the environment variables string from the JSON response
+                  ENV_VARS_STRING=$(echo "$ENV_VARS" | jq -r '.environmentVariables')
+
+                  # Add 'export' to each line
+                  echo "$ENV_VARS_STRING" | sed 's/^/export /' > ~/project/environment_variables.txt
+
+      - run:
```

Comment:
we should remove this right before merge

Reply:
yep